## Investigation

In [2]:
%run './lib/libraries.ipynb'

dython              0.6.1
tinydb              4.2.0


In [2]:
dfRaw  = load_df('dfRaw')

In [3]:
list_todo()


{'todo': 'Investigate 3298 null values in response_correctness', 'finished': True}
{'todo': 'Investigate 1566 [unassigned] in response_correctness', 'finished': True}
{'todo': 'Investigate null values in item_type_code_name', 'finished': False}
{'todo': 'Reduce number of levels in item_type_code_name', 'finished': False}
{'todo': 'Remove unary learner_attempt_status = "fully scored"', 'finished': True}
{'todo': 'Investigate number_of_learners > 40 outliers', 'finished': False}
{'todo': 'Investigate binary variables assignment_attempt_number and assignment_max_attempts', 'finished': True}
{'todo': 'Investigate assignment_attempt_number and assignment_max_attempts both have 1566 values', 'finished': True}
{'todo': 'Investigate final_score_unweighted has 16,582 zero values', 'finished': False}
{'todo': 'Replace default dates with nulls', 'finished': False}
{'todo': 'Investigate why some dates are null', 'finished': False}
{'todo': 'Remove variables assignment_late_submission, is_manual_sc

### [Investigate binary variables assignment_attempt_number and assignment_max_attempts](./investigation/attempt-number-vs-max.ipynb)

### [Investigate 1566 [unassigned] in response_correctness](./investigation/1566-unassigned-in-response-correctness.ipynb)

### [Investigate 3298 null values in response_correctness](./investigation/3298-null-in-response-correctness.ipynb)

### [Investigate null values in item_type_code_name](./investigation/item-type-code-name-nulls.ipynb)

### [Investigate number_of_learners > 40 outliers](./investigation/num-learners-over-40.ipynb)

In [4]:
list_todo()

- [Investigate number_of_learners > 40 outliers](./investigation/num-learners-over-40.ipynb)

{'todo': 'Investigate 3298 null values in response_correctness', 'finished': True}
{'todo': 'Investigate 1566 [unassigned] in response_correctness', 'finished': True}
{'todo': 'Investigate null values in item_type_code_name', 'finished': True}
{'todo': 'Reduce number of levels in item_type_code_name', 'finished': False}
{'todo': 'Remove unary learner_attempt_status = "fully scored"', 'finished': True}
{'todo': 'Investigate number_of_learners > 40 outliers', 'finished': False}
{'todo': 'Investigate binary variables assignment_attempt_number and assignment_max_attempts', 'finished': True}
{'todo': 'Investigate assignment_attempt_number and assignment_max_attempts both have 1566 values', 'finished': True}
{'todo': 'Investigate final_score_unweighted has 16,582 zero values', 'finished': False}
{'todo': 'Replace default dates with nulls', 'finished': False}
{'todo': 'Investigate why some dates are null', 'finished': False}
{'todo': 'Remove variables assignment_late_submission, is_manual_sco

###### Statistics of final_score_unweighted to item_type_code_name = Null

In [ ]:
dfInv.filter(F.col("item_type_code_name").isNull() == True).select("final_score_unweighted").toPandas().hist(bins=60)

- Strongly correlated to a final_score_unweighted of 0

In [ ]:
finish_todo('Investigate null values in item_type_code_name')


